config/v1/types_cluster_operator.go

```diff
@@ -0,0 +1,106 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// ClusterOperatorList is a list of OperatorStatus resources.
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterOperatorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ClusterOperator `json:"items"`
+}
+
+// ClusterOperator is the Custom Resource object which holds the current state
+// of an operator. This object is used by operators to convey their state to
+// the rest of the cluster.
+// +genclient
```

Comment:
I think this should be cluster cluster-scoped.  A namespace scoped **Cluster**Operator seems weird.

Reply:
the advantage is deleting the namespace deletes the operator, which means that you don't have to give an operator the rights to edit any other cluster operator.

Reply:
> the advantage is deleting the namespace deletes the operator, which means that you don't have to give an operator the rights to edit any other cluster operator.

These are already fairly privileged things.  Having them at the namespace level means, "sure you want foo, but in which namespace".  It's a question that I don't think we need to be asking ourselves.

In addition, you could choose to have the CVO create it and then only grant update access.

Reply:
David mostly convinced me to make it globally scoped so he can avoid having to type `--all-namespaces`.

It's not any less secure.  It does lose some of the "cleanup" aspects (deleting the namespace would remove the status, whereas now you would have bits left over), but even today we have bits outside the namespace like cluster roles.